### ISSUE TITLE
[BUG] animations give `read: Connection reset by peer` and `send: Broken pipe`

### ISSUE THREAD

**Issue by useredsa** [JuliaPlots/Plots.jl#3012]
### Details

Using Arch linux (tested on 2 different computers):

`sudo pacman -S julia`
`inside julia: add Plots`

Result in animation not working. This bug can be tested using the [Lorenz Attractor Example](http://docs.juliaplots.org/latest/#simple-is-beautiful).

Plotting works without issues and I've checked that the dependencies for GR are installed in my system.

However, when trying to use `@gif` or `@animation` or a manual approach, I get the errors:

```
read: Connection reset by peer
send: Broken pipe
```

repeatedly and "broken" windows that spawn and close automatically with each message.

### Backends

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |  x  |     |
pyplot       |     |     | x
plotly       |     |     | x
plotlyjs     |     |     | x
pgfplotsx    |     |     | x
inspectdr    |     |     | x

### Versions

Plots.jl version:
  [91a5bcdd] Plots v1.6.6 (I think this is what I need to put)
Backend  version (`]st -m`):
  [28b8d3ca] GR v0.52.0
Output of `versioninfo()`:

```
Julia Version 1.5.2
Commit 539f3ce943* (2020-09-23 23:17 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-10.0.1 (ORCJIT, skylake)
```

**Comment by useredsa** [comment]
With respect to this problem, I have found that if I include `IJulia` (`using IJulia`), then the plot doesn't appear, for example when writing `plot([1,2,3], [1,2,3])`, not even with `display(plot(...))`, but the `@gif` and `@animate` macros can therefore run without issues and the result gets saved in a file I can preview with another application.

**Comment by gustaphe** [comment]
Can replicate Also on Arch Linux btw, using i3 window manager. No problem generating files, but I get these warnings and windows flashing quickly by while plotting.

**Comment by useredsa** [comment]
> Can replicate. Also on Arch Linux btw, using i3 window manager. No problem generating files, but I get these warnings and windows flashing quickly by while plotting.

I can tell t also happens in Ubuntu, at least in some versions.

**Comment by elrond2001** [comment]
Manjaro KDE has the same problem. After `using IJulia` no error is reported and no plot is shown.
output of `versioninfo()`
```
Julia Version 1.5.3
Commit 788b2c77c1 (2020-11-09 13:37 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)
Environment:
  JULIA_PKG_SERVER = https://mirrors.tuna.tsinghua.edu.cn/julia
```

**Comment by useredsa** [comment]
@elrond2001, at least I have been able to save the result to a file and open it in my image/gif viewer. It is enough to work I believe.

I believe everything works correctly in atom+juno, so that's another option.

**Comment by elrond2001** [comment]
@useredsa I just test it in Pluto, everything works fine. Maybe I should move on and not thinking of seeing a window of gif  :D

**Comment by henry2004y** [comment]
Also had this issue with Julia 1.6 on Ubuntu 18.04

**Comment by umbriquse** [comment]
I am also having this issue. In my case its only sporadically that I get the `read: Connection reset by peer` and `send: Broken Pipe` while animating a 3D plot. I am also getting the plotted frames that are automatically being opened and then closed for each frame in the animation.

versioninfo()
```
Julia Version 1.6.0
Commit f9720dc2eb (2021-03-24 12:55 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin19.6.0)
  CPU: Apple M1
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, westmere)
```
Package Version:
```
pkg> st Plots
> Plots v1.12.0
```

**Comment by akawashiro** [comment]
I have also the same problem with my Arch linux.

```
julia> versioninfo()
Julia Version 1.6.1
Commit 6aaedecc44* (2021-04-23 05:59 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.1.0 (ORCJIT, skylake)
  
  (@v1.6) pkg> st Plots
      Status `~/.julia/environments/v1.6/Project.toml`
  [91a5bcdd] Plots v1.12.0
  
  > uname -srm
Linux 5.11.16-arch1-1 x86_64
  ```

**Comment by diederikb** [comment]
Setting `ENV["GKSwstype"] = "nul"` in the code prevented this for me. This was actually a solution for another problem discussed [here](https://discourse.julialang.org/t/unable-to-display-plot-using-the-repl-gks-errors/12826/3).

**Comment by useredsa** [comment]
> Setting `ENV["GKSwstype"] = "nul"` in the code prevented this for me. This was actually a solution for another problem discussed [here](https://discourse.julialang.org/t/unable-to-display-plot-using-the-repl-gks-errors/12826/3).

Yes, but then you are unavailable to view the results from plotting automatically. You have to open the file created by julia.

**Comment by Libbum** [comment]
Any particular reason why this was closed? It's still an ongoing issue with no appropriate work around.

**Comment by t-bltg** [comment]
> Any particular reason why this was closed? It's still an ongoing issue with no appropriate work around.

I've just closed `60+` (old) issues in a row, so some might still be relevant.

However for this issue, there is a workaround: setting `ENV["GKSwstype"] = "nul"`.
I believe this is a `GR` issue, not `Plots`.

**Comment by useredsa** [comment]
> However for this issue, there is a workaround: setting `ENV["GKSwstype"] = "nul"`.

Well, only if you don't mind having to open a file instead of using the library as everyone else.



> I believe this is a `GR` issue, not `Plots`.

I think it happens with other backends.

**Comment by t-bltg** [comment]
> I think it happens with other backends.

Please provide a mwe.
